x-pack/plugins/task_manager/server/queries/mark_available_tasks_as_claimed.ts

```diff
@@ -128,7 +128,9 @@ function getSortByPriority(definitions: TaskTypeDictionary): estypes.SortCombina
         // TODO: we could do this locally as well, but they may starve
         source: `
           String taskType = doc['task.taskType'].value;
-          if (params.priority_map.containsKey(taskType)) {
+          if (doc['task.priority'].size() != 0) {
+            return doc['task.priority'].value;
+          } else if (params.priority_map.containsKey(taskType)) {
```

Comment:
Moving the conversation from https://github.com/elastic/kibana/pull/199218#discussion_r1846616085 here, sorry for the delay!

>Either would work! The way it currently is, we would sort by priority=descending which makes more sense in my head but definitely an argument for the reverse

Hmm, is it easy to check what other queueing products like RabbitMQ, Kafka, etc use for priority order? We can see if they do asc or desc and follow suit.